src/rust/ensogl/lib/core/src/system/gpu/data/attribute.rs

```diff
@@ -129,7 +142,7 @@ impl {
             for buffer in &self.buffers {
                 buffer.set_to_default(id.into())
             }
-            self.free_ids.push(id);
+            self.free_ids.insert(*id);
```

Comment:
We replace popping and pushing with asymptotic complexity O(1) with removing/inserting having complexity O(n). Have you checked if it alter our IDE performance? I believe that it may hurt us when we create really many nodes, then remove the first one and add again.

And I believe that we want to guarantee the order of "consecutively added sprites" to fix the issue in topic - so this guarantee should be expressed in docs, and/or in comment here.